libnd4j/include/array/ShapeDescriptor.h

```diff
@@ -74,6 +66,14 @@ class SD_LIB_EXPORT ShapeDescriptor {
   }
 
  public:
+  int _rank = 0;
+  std::vector<sd::LongType> _shape_strides;
+  // std::vector<sd::LongType> _strides;
+  sd::LongType _ews = 1;
+  char _order = 'c';
+  DataType _dataType;
+  sd::LongType _extraProperties = 0;
+  sd::LongType _paddedAllocSize = 0;
```

Comment:
Does this have to be public? Or is that left over from debugging?

Reply:
Yes. This was orignally private but I ended up needing to use extraProperties in a  few places.